### PR TITLE
Added choice picker in settings and game settings sheet to switch the clock position to the left side

### PR DESCRIPTION
**Pull request by Jimima**
Related to issue https://github.com/lichess-org/mobile/issues/1098

New developer here, I have had a crack at this. If there are omissions or errors or anything I have done incorrectly please just let me know. I am happy to learn more 😄

### CODE REVIEW

**Review by veloce — COMMENTED** [review]
Thanks for this PR. Just a couple of comments, otherwise it looks good.

File: lib/src/model/settings/board_preferences.dart [hunk 1]
```diff
@@ -106,6 +112,7 @@ class BoardPrefs with _$BoardPrefs implements Serializable {
     required bool coordinates,
     required bool pieceAnimation,
     required bool showMaterialDifference,
+    required bool switchClockPosition,
```
[the remark this answers is not in the document]
Comment (Jimima) [reply lib/src/model/settings/board_preferences.dart:115]: Hey, I agree this makes more sense. How would you approach the UI in that case? Would you keep a toggle or change to a choice picker similar to the theme selector? Or something else?

[the remark this answers is not in the document]
Comment (veloce) [reply lib/src/model/settings/board_preferences.dart:115]: Choice picker looks good in that case yes.

File: lib/src/model/settings/board_preferences.dart [hunk 2]
```diff
@@ -132,6 +139,7 @@ class BoardPrefs with _$BoardPrefs implements Serializable {
     coordinates: true,
     pieceAnimation: true,
     showMaterialDifference: true,
+    clockPosition: ClockPosition.left,
```
Comment (tom-anders) [inline lib/src/model/settings/board_preferences.dart:142]: Shouldn't `right` be the default to match the current UI?

**Comment by tom-anders** [comment]
Currently, when the clock is set to `left`, there's currently some space on the right of the player widget (see screenshot below).

I think you can fix this by adding `mainAxisAlignment: clockPosition == ClockPosition.right ? MainAxisAlignment.start : MainAxisAlignment.end` to the two `Row` widgets in side the `playeWidget`

Screenshot_1731145252

File: lib/src/model/settings/board_preferences.dart (hunk 2 above)
Comment (Jimima) [reply lib/src/model/settings/board_preferences.dart:142]: Yes typo will fix

**Comment by tom-anders** [comment]
Yeah look good! But I think you also need it to the `Row` that displays the material difference below the user name (see my original screenshot, you can capture some pieces to check this)

**Comment by Jimima** [comment]
Yes I seem, makes sense I should have made some captures! Now added.

File: lib/src/model/settings/board_preferences.dart (hunk 2 above)
Comment (Jimima) [reply lib/src/model/settings/board_preferences.dart]: Fixed

**Review by veloce — APPROVED** [review]

**Comment by Jimima** [comment]
😁 thanks!

**Pull request merged**